File: frontend/src/types/recruiter.ts
import type { ApplicationTimelineEvent } from '@/types/applications';

export const PIPELINE_STAGES = ['Applied', 'Screening', 'Shortlisted', 'Technical Interview', 'HR Interview', 'Final Review', 'Offer', 'Hired', 'Rejected'] as const;
export type PipelineStage = (typeof PIPELINE_STAGES)[number];
export type RecruiterNote = { id: string; author: string; content: string; created_at: string };
export type RecruiterCandidate = {
  id: string; application_id: string; name: string; email: string; phone: string;
  location: string; headline: string; experience: string; education: string[];
  certifications: string[]; skills: string[]; portfolio_url: string | null;
  github_url: string | null; linkedin_url: string | null; job_id: string;
  job_title: string; status: PipelineStage; rating: number; tags: string[];
  bookmarked: boolean; resume_name: string; cover_letter: string; applied_at: string;
  notes: RecruiterNote[]; timeline: ApplicationTimelineEvent[];
};
export type RecruiterCandidateList = { candidates: RecruiterCandidate[]; total: number };
export type RecruiterCandidateUpdate = { candidate_id: string; message: string; candidate: RecruiterCandidate };

File: backend/app/schemas/recruiter.py
from typing import Literal

from pydantic import BaseModel, Field, HttpUrl

from app.schemas.applications import ApplicationTimelineEvent

PipelineStage = Literal[
    "Applied",
    "Screening",
    "Shortlisted",
    "Technical Interview",
    "HR Interview",
    "Final Review",
    "Offer",
    "Hired",
    "Rejected",
]


class RecruiterNote(BaseModel):
    id: str
    author: str
    content: str
    created_at: str


class RecruiterCandidate(BaseModel):
    id: str
    application_id: str
    name: str
    email: str
    phone: str
    location: str
    headline: str
    experience: str
    education: list[str]
    certifications: list[str]
    skills: list[str]
    portfolio_url: HttpUrl | None = None
    github_url: HttpUrl | None = None
    linkedin_url: HttpUrl | None = None
    job_id: str
    job_title: str
    status: PipelineStage
    rating: int = Field(ge=1, le=5)
    tags: list[str] = Field(default_factory=list)
    bookmarked: bool = False
    resume_name: str
    cover_letter: str
    applied_at: str
    notes: list[RecruiterNote] = Field(default_factory=list)
    timeline: list[ApplicationTimelineEvent] = Field(default_factory=list)


class RecruiterCandidateList(BaseModel):
    candidates: list[RecruiterCandidate]
    total: int


class RecruiterStatusPayload(BaseModel):
    status: PipelineStage


class RecruiterNotePayload(BaseModel):
    content: str = Field(min_length=2, max_length=2000)


class RecruiterRatingPayload(BaseModel):
    rating: int = Field(ge=1, le=5)


class RecruiterCandidateUpdate(BaseModel):
    candidate_id: str
    message: str
    candidate: RecruiterCandidate
